src/components/popover/popover.component.ts

```diff
@@ -134,20 +135,24 @@ export class PopoverComponent implements OnChanges {
     }
 
     onRemapKey(): void {
-        try {
-            let keyAction = this.selectedTab.toKeyAction();
-            this.remap.emit(keyAction);
-        } catch (e) {
-            // TODO: show error dialog
-            console.error(e);
+        if (this.isValidAction) {
+            try {
+                let keyAction = this.selectedTab.toKeyAction();
+                this.remap.emit(keyAction);
+            } catch (e) {
+                // TODO: show error dialog
+                console.error(e);
+            }
         }
     }
 
-    @HostListener('keydown.escape') onEscape(): void {
+    @HostListener('keydown.escape')
+    onEscape(): void {
         this.cancel.emit();
     }
 
     selectTab(tab: TabName): void {
+        this.isValidAction = !(tab === this.tabName.Keypress || tab === this.tabName.Mouse);
```

Comment:
I know, but I think it is the tabs responsibility when it find the key action valid, so we shouln't assume anything here.